From 726cf412ad28f0045a7b8b507b3e01b51972ea0d Mon Sep 17 00:00:00 2001
From: Sasha Levin <sashal@kernel.org>
Date: Wed, 12 Aug 2026 10:51:47 +0200
Subject: swapfile: call cond_resched() before locking si->lock

From: Guillaume Morin <guillaume@morinfr.org>

The 6.12 backport of commit 66366d291f66 ("mm/swap: add cond_resched()
in swap_reclaim_full_clusters to prevent softlockup"),
added a cond_resched() after locking si->lock. Move cond_resched()
before spin_lock().

Signed-off-by: Guillaume Morin <guillaume@morinfr.org>
Fixes: 60cbe67d1342 ("mm/swap: add cond_resched() in swap_reclaim_full_clusters to prevent softlockup")
Link: https://lore.kernel.org/stable/ansgKOOXi6KE3oKf@bender.morinfr.org/
Cc: stable@vger.kernel.org
Cc: Chris Li <chrisl@kernel.org>
Cc: Zijiang Huang <kerayhuang@tencent.com>
Cc: gregkh@linuxfoundation.org
Signed-off-by: Sasha Levin <sashal@kernel.org>
---
 mm/swapfile.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/swapfile.c b/mm/swapfile.c
index 552df432c37f4..ed0a5ed25e65b 100644
--- a/mm/swapfile.c
+++ b/mm/swapfile.c
@@ -770,11 +770,11 @@ static void swap_reclaim_full_clusters(struct swap_info_struct *si, bool force)
 			}
 			offset++;
 		}
+		cond_resched();
 		spin_lock(&si->lock);
 
 		if (to_scan <= 0)
 			break;
-		cond_resched();
 	}
 }
 
-- 
2.53.0

